bde/ml/models.py

```diff
@@ -188,18 +347,32 @@ def fit(
         :param y: The labels. If y is None, X is assumed to include the labels as well.
         :return: The fitted estimator.
         """
-        self.params_ = None
-        self.history_ = dict()
-        if self.metrics is None:
-            self.metrics = list()
-        else:
+        if y is None:
+            y = X
+        bde.utils.utils.check_fit_input(X, y)
+        metrics = [] if self.metrics is None else self.metrics
+        if len(metrics) > 0:
             raise ValueError(f"Metrics are not yet supported.")  # TODO: Remove after implementation
```

Comment:
rather not implemented yet error :)

Reply:
Implemented in 9aab1cd421656f51cb794a001351fb2e2011fb52.